libs/s25main/world/GameWorldGame.cpp

```diff
@@ -722,7 +722,8 @@ void GameWorldGame::Attack(const unsigned char player_attacker, const MapPoint p
                            const bool strong_soldiers)
 {
     auto* attacked_building = GetSpecObj<nobBaseMilitary>(pt);
-    if(!attacked_building || !attacked_building->IsAttackable(player_attacker))
+    if(!attacked_building || !attacked_building->IsAttackable(player_attacker)
+       || (GetGGS().getSelection(AddonId::PEACEFUL)))
```

Comment:
this should not be necessary if you simply disable the attack / sea attack tabs

Reply:
Thank you for all your suggestions! I think it is important that in the game logic itself attacks are prohibited (and not only in the UI), because otherwise the AI will happily attack me while I can't attack it. But these reviews made me realize that IsAttackable is directly or indirectly called in all the places I checked for the peaceful mode, so I replaced all these checks with a single one in nobBaseMilitary::IsAttackable, which is more elegant, I hope you agree.